### PR TITLE
Task Js DOM

### PR DESCRIPTION
**Pull request by OlexiyDobroskok**
### Document Object Model

[Demo](https://olexiydobroskok.github.io/JS__DOM/) |
[Code base](https://github.com/OlexiyDobroskok/JS__DOM)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+aside.className = "sidebar";
+main.prepend(aside);
+aside.prepend(...addButtons());
+
+function addButtons() {
+  const buttonArr = [];
+  for (let i = 0; i < 9; i++) {
+    let button = document.createElement("button");
+    button.className = "sidebar__button";
+    button.id = "dt" + [i];
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:101]: Try to use methods of an array (map, reduce, forEach)

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    about:
+      "We join Roland and his ka-tet as a ferocious storm halts their progress along the Path of the Beam. As they shelter from the screaming wind and snapping trees, Roland tells them not just one strange tale, but two – and in doing so sheds fascinating light on his own troubled past.",
+    photo: "./img/The Wind through the Keyhole.jpg",
+    alt: "Illustration for Stephen King's The Dark Tower: The Wind Through the Keyhole",
+  },
+];
+
+let main = document.querySelector(".content");
+let aside = document.createElement("aside");
+aside.className = "sidebar";
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:92]: The variables must be as constant.

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    title: "The Dark Tower: The Wind Through the Keyhole",
+    quote: "“Balls to your sorry ... Ka works and the world moves on. ”",
+    about:
+      "We join Roland and his ka-tet as a ferocious storm halts their progress along the Path of the Beam. As they shelter from the screaming wind and snapping trees, Roland tells them not just one strange tale, but two – and in doing so sheds fascinating light on his own troubled past.",
+    photo: "./img/The Wind through the Keyhole.jpg",
+    alt: "Illustration for Stephen King's The Dark Tower: The Wind Through the Keyhole",
+  },
+];
+
+let main = document.querySelector(".content");
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:90]: Set better naming for variables.
`mainList, btnList ......`

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    if (eventBtn.id === darkTowerSeries[i].id) {
+      h2.innerHTML = darkTowerSeries[i].title;
+      paragraphQuote.innerHTML = darkTowerSeries[i].quote;
+      paragraph.innerHTML = darkTowerSeries[i].about;
+      img.src = darkTowerSeries[i].photo;
+      img.alt = darkTowerSeries[i].alt;
+    }
+  }
+}
+contentCreator(dt0);
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:146]: Create an initial variable.

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+div.after(img);
+
+aside.onclick = function (event) {
+  let target = event.target;
+  if (target.tagName !== "BUTTON") return;
+  contentCreator(target);
+};
+
+function contentCreator(eventBtn) {
+  for (let i = 0; i < darkTowerSeries.length; i++) {
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:136]: I would recommend adding all variables which you create in this function.
Your function gets a parameter, it is good. Try to use a method of array `find` or `filter` to get needed item from array list.

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+paragraphQuote.className = "quote__text";
+blockquote.prepend(paragraphQuote);
+let paragraph = document.createElement("p");
+paragraph.className = "content__text";
+blockquote.after(paragraph);
+let img = document.createElement("img");
+img.className = "content__img";
+div.after(img);
+
+aside.onclick = function (event) {
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:129]: Use [event delegation](https://javascript.info/event-delegation)

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See some comments below.

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  const contentData = darkTowerSeries.filter((el) => eventBtn.id === el.id);
+  contentTitle.innerHTML = contentData[0].title;
+  paragraphQuote.innerHTML = contentData[0].quote;
+  paragraph.innerHTML = contentData[0].about;
+  bookImage.src = contentData[0].photo;
+  bookImage.alt = contentData[0].alt;
+}
+
+sidebar.addEventListener("click", function (event) {
+  let target = event.target;
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:139]: Must be `const`

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+contentContainer.prepend(sidebar);
+sidebar.prepend(...addButtons());
+
+function addButtons() {
+  const buttonList = [];
+  darkTowerSeries.forEach((el, index) => {
+    const button = document.createElement("button");
+    button.className = "sidebar__button";
+    button.id = "dt" + index;
+    button.innerHTML = darkTowerSeries[index].title;
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:102]: What is the mean name 'el'? It is better named as 'serie'
```suggestion
    button.innerHTML = el.title;
```

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+const bookImage = document.createElement("img");
+bookImage.className = "content__img";
+textContentContainer.after(bookImage);
+
+function contentCreator(eventBtn) {
+  const contentData = darkTowerSeries.filter((el) => eventBtn.id === el.id);
+  contentTitle.innerHTML = contentData[0].title;
+  paragraphQuote.innerHTML = contentData[0].quote;
+  paragraph.innerHTML = contentData[0].about;
+  bookImage.src = contentData[0].photo;
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:134]: Read about [destruction](https://javascript.info/destructuring-assignment)

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Add a background style for the text. Because the letters blend into the background image

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+const sidebar = document.createElement("aside");
+sidebar.className = "sidebar";
+contentContainer.prepend(sidebar);
+sidebar.prepend(...addButtons());
+
+function addButtons() {
+  const buttonList = [];
+  darkTowerSeries.forEach((serie, index) => {
+    const button = document.createElement("button");
+    button.className = "sidebar__button";
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:100]: Read about [Array.map](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and [array destructuring](https://javascript.info/destructuring-assignment#array-destructuring).
Then you can write your code more simply.

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+blockquote.after(paragraph);
+const bookImage = document.createElement("img");
+bookImage.className = "content__img";
+textContentContainer.after(bookImage);
+
+function contentCreator(eventBtn) {
+  const contentData = darkTowerSeries.filter(
+    (serie) => eventBtn.id === serie.id
+  );
+  const [serie] = contentData;
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:133]: It is not necessary. You are using a filter that has a search result by id.

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+  console.log(serie);
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:134]: Delete debug line.

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+sidebar.addEventListener("click", function (event) {
+  const target = event.target;
+  if (target.closest(".sidebar__button")) {
+    contentCreator(target);
+  }
+});
+
+const frontPage = contentCreator(dt0);
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:149]: Create initial variable with value 'dt0';

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+
+sidebar.addEventListener("click", function (event) {
+  const target = event.target;
+  if (target.closest(".sidebar__button")) {
+    contentCreator(target);
+  }
+});
+
+const aboutSeries = dt0;
+const frontPage = contentCreator(aboutSeries);
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:145]:
```suggestion
contentCreator(aboutSeries);
```

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+function contentCreator(eventBtn) {
+  const [serie] = darkTowerSeries.filter((serie) => eventBtn.id === serie.id);
+  contentTitle.innerHTML = serie.title;
+  paragraphQuote.innerHTML = serie.quote;
+  paragraph.innerHTML = serie.about;
+  bookImage.src = serie.photo;
+  bookImage.alt = serie.alt;
+}
+
+sidebar.addEventListener("click", function (event) {
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:137]:
```suggestion
sidebar.addEventListener("click", function ({ target }) {
```

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+blockquote.prepend(paragraphQuote);
+const paragraph = document.createElement("p");
+paragraph.className = "content__text";
+blockquote.after(paragraph);
+const bookImage = document.createElement("img");
+bookImage.className = "content__img";
+textContentContainer.after(bookImage);
+
+function contentCreator(eventBtn) {
+  const [serie] = darkTowerSeries.filter((serie) => eventBtn.id === serie.id);
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:129]:
```suggestion
  const contentData = darkTowerSeries.filter((serie) => eventBtn.id === serie.id);
  const [{ title, quote, about, photo, alt }] = contentData;
```

File: submissions/OlexiyDobroskok/task_js_dom/script.js [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+
+function addButtons() {
+  const buttonList = darkTowerSeries.map((serie, serieNumber) => {
+    const button = document.createElement("button");
+    button.className = "sidebar__button";
+    button.id = "dt" + serieNumber;
+    button.innerHTML = darkTowerSeries[serieNumber].title;
+    return button;
+  });
+  return buttonList;
```
Comment (madmaxWMFU) [inline submissions/OlexiyDobroskok/task_js_dom/script.js:104]:
```suggestion
    return darkTowerSeries.map(serie => {
      const { id, title } = serie;
      const btn = document.createElement('button');
      btn.classList.add('sidebar__button');
      btn.id = id;
      btn.innerText = title;
      return btn;
    });
```

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

**Comment by OlexiyDobroskok** [comment]
@madmaxWMFU thanks for the help!

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**